File: sentimental-mario-less/mario.py
def main():
    height = 0
    while (height < 1 or height > 8):
        try:
            height = int(input("La hauteur de la piramide entier entre 1 et 8 :"))
        except:
            pass

    for i in range(height):
        nb_espace = height - 1 - i
        nb_hashes = i + 1
        print_row(nb_hashes, nb_espace)


def print_row(nb_hashes, nb_espace):
    print(" " * nb_espace + "#" * nb_hashes)


main()

File: mario/mario.c
#include <stdio.h>
#include <cs50.h>

void print_row(int nb_espace, int nb_hashes);

int main(void)
{
    int height;
    do
    {
        // demande la taille height
        height = get_int("Quel est la taille de la piramide ?");
    }
    while (height < 1);


    for (int i = 0; i < height; i++)
    {
        int nb_hashes = height - 1;
        int nb_espace = height - i;
        print_row(nb_espace, nb_hashes);
    }
}


void print_row(int nb_espace, int nb_hashes)
{
    for (int i = 0; i < nb_espace; i++)
    {
        printf(" ");
    }
    for (int i = 0; i < nb_hashes; i++)
    {
        printf("#");
    }
    printf("\n");
}
